metrics/super_glue/super_glue.py

```diff
@@ -0,0 +1,189 @@
+  journal={arXiv preprint arXiv:1905.00537},
+  year={2019}
+}
+"""
+
+_DESCRIPTION = """\
+SuperGLUE (https://super.gluebenchmark.com/) is a new benchmark styled after
+GLUE with a new set of more difficult language understanding tasks, improved
+resources, and a new public leaderboard.
+"""
+
+_KWARGS_DESCRIPTION = """
+Compute SuperGLUE evaluation metric associated to each SuperGLUE dataset.
+Args:
+    predictions: list of predictions to score. Depending on the SuperGlUE subset:
+        - for 'record': list of question-answer dictionaries with the following keys:
+            - 'idx': index of the question as specified by the dataset
+            - 'prediction_text': the predicted answer text
+        - for 'multirc': list of question-answer dictionaries with the following keys:
+            - 'idx': index of the question-answer pair as specified by the dataset
+            - 'prediction': the predicted answer text
+        - otherwise: list of predicted labels
+    references: list of reference labels. Depending on the SuperGLUE subset:
+        - for 'record': list of question-answers dictionaries with the following keys:
+            - 'idx': index of the question as specified by the dataset
+            - 'answers': list of possible answers
+        - otherwise: list of reference labels
+Returns: depending on the SuperGLUE subset, one or several of:
+    "accuracy": Accuracy
+    "f1": F1 score
+    "exact_match": Exact match between answer and gold answer
+    "matthews_correlation": Matthew Correlation
+"""
+
+
+def simple_accuracy(preds, labels):
+    return (preds == labels).mean()
+
+
+def acc_and_f1(preds, labels, f1_avg="binary"):
+    acc = simple_accuracy(preds, labels)
+    f1 = f1_score(y_true=labels, y_pred=preds, average=f1_avg)
+    return {
+        "accuracy": acc,
+        "f1": f1,
+    }
+
+
+def evaluate_multirc(ids_preds, labels):
+    """
+    Computes F1 score and Exact Match for MultiRC predictions.
+    """
+    question_map = {}
+    for id_pred, label in zip(ids_preds, labels):
+        question_id = "{}-{}".format(id_pred["idx"]["paragraph"], id_pred["idx"]["question"])
+        pred = id_pred["prediction"]
+        if question_id in question_map:
+            question_map[question_id].append((pred, label))
+        else:
+            question_map[question_id] = [(pred, label)]
+    f1s, ems = [], []
+    for question, preds_labels in question_map.items():
+        question_preds, question_labels = zip(*preds_labels)
+        f1 = f1_score(y_true=question_labels, y_pred=question_preds)
+        f1s.append(f1)
+        em = int(sum([p == l for p, l in preds_labels]) == len(preds_labels))
+        ems.append(em)
+    avg_f1 = sum(f1s) / len(f1s)
+    avg_em = sum(ems) / len(ems)
+    return {"f1": avg_f1, "exact_match": avg_em}
+
+
+class SuperGlue(datasets.Metric):
```

Comment:
```suggestion
@datasets.utils.file_utils.add_start_docstrings(_DESCRIPTION, _KWARGS_DESCRIPTION)
class SuperGlue(datasets.Metric):
```
Could you also add the kwargs this way please ?